src/components/__tests__/DisciplineGroup.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import '@testing-library/jest-dom/extend-expect'
+import { render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import DisciplineGroup from '../DisciplineGroup'
+
+const defaultTypes = {
+  trad: true,
+  sport: true,
+  tr: true,
+  bouldering: true
+}
+let climbTypes
+const setClimbTypes = (value): void => {
+  climbTypes = value
+}
+
+describe('DisciplineGroup', () => {
+  beforeEach(() => {
+    climbTypes = {
+      trad: true,
+      sport: true,
+      tr: true,
+      bouldering: true
+    }
+  })
+  it('renders component without error', () => {
+    render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+  })
+
+  it('all 4 types selected by default', () => {
+    const { getByRole } = render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+    const sportButton = getByRole('checkbox', { name: /Sport/i })
+    const tradButton = getByRole('checkbox', { name: /Trad/i })
+    const topRopeButton = getByRole('checkbox', { name: /Top rope/i })
+
+    expect(sportButton).toBeChecked()
+    expect(tradButton).toBeChecked()
+    expect(topRopeButton).toBeChecked()
+  })
+
+  it('changes values and resets to defaultTypes when component regenerated', async () => {
+    const clickButton = async (button): Promise<void> => {
+      await userEvent.click(button)
+      rerender(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+    }
+
+    const { container, getByRole, rerender, unmount } = render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+    expect(container.getElementsByClassName('border-neutral-800').length).toBe(4)
+
+    const sportButton = getByRole('checkbox', { name: /Sport/i })
+    const tradButton = getByRole('checkbox', { name: /Trad/i })
+    const topRopeButton = getByRole('checkbox', { name: /Top rope/i })
```

Comment:
can you still get by role button but expect .toBeChecked?
that was what the comment on the pr mentioned.

did you find out get by checkbox to be best practices when researching?

Reply:
cannot .getByRole('button') because button is not a type here, checkbox is. the ask was to change role to checkbox. i figured it was a small typo in declaring the new sportButton variable in the comment above. 
```javascript
const sportButton = getByRole('button', { name: /Sport/i })
// should have been
const sportButton = getByRole('checkbox', { name: /Sport/i })
```

regarding best practices, i didn't find anything that argued against using this as a way to get the button.  [getByTestId](https://testing-library.com/docs/queries/bytestid/) might also be good

Reply:
get by test ID is frowned upon.